.github/workflows/main-build.yml

```diff
@@ -0,0 +1,79 @@
+name: main-build
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - v*.*.*
+  pull_request:
+    branches:
+      - main
+
+env:
+  VERSION_TAG: experimental
+
+jobs:
+  build:
+    name: Build source and create Docker images
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Setup dotnet environment
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 3.1.x
+
+      - name: Install dependencies
+        run: dotnet restore
+
+      - name: Build solution
+        run: dotnet build --configuration Release --no-restore
+
+      - name: Run unit tests
+        run: dotnet test --no-restore
+
+      - name: Build Docker image for order-generator
+        uses: docker/build-push-action@v2
+        with:
+          file: ./Keda.CosmosDbScaler.Demo.OrderGenerator/Dockerfile
+          tags: cosmosdb-order-generator:${{ env.VERSION_TAG }}
+
+      - name: Build Docker image for order-processor
+        uses: docker/build-push-action@v2
+        with:
+          file: ./Keda.CosmosDbScaler.Demo.OrderProcessor/Dockerfile
+          tags: cosmosdb-order-processor:${{ env.VERSION_TAG }}
+
+      - name: Login to GitHub container registry
+        if: github.event_name == 'push'
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract Docker metadata
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: ghcr.io/${{ github.repository }}
+          tags: |
+            type=ref,event=pr
+            type=ref,event=tag
+            type=raw,value=${{ env.VERSION_TAG }}
```

Comment:
A new tag, along with creating versioned tag for the GHCR image, will also override the `experimental` tag. I will look into this.

Reply:
Resolved to some extent. There will be no image with tag `latest` on new repo tag. It should be okay to live with overriding behavior of `experimental` tag for now, since in most cases, the latest commit on `main` branch will be tagged.